core/src/main/codegen/templates/Parser.jj

```diff
@@ -1272,6 +1274,36 @@ SqlNode TableRefWithHintsOpt() :
     }
 }
 
+/**
+ * Parses a target table reference with optional hints.
```

Comment:
Actually, since you've defined a way to create a table identifier from another identifier `SqlTableIdentifierWithID(SqlIdentifier id)`, couldn't you just call TableRefWithHintsOpt and convert the value? I feel like that would be cleaner than doing this copy paste.

Reply:
Or are you not, because we need a the table identifier to be a TargetTableIdentifier?

Reply:
I'll update the comment. I don't want to convert the value after parsing because I would have to handle the hint case as well. Since we don't have tests for that I want to hold off on those changes. I will update the comments.